spec/copy_branch_protection_action_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+require 'tmpdir'
+
+describe Fastlane::Actions::CopyBranchProtectionAction do
+  let(:repo) { 'automattic/demorepo' }
+  let(:from_branch) { 'trunk' }
+  let(:to_branch) { 'release/12.3' }
+  let(:github_token) { 'stubbed-gh-token' }
+  let(:client) do
+    instance_double(
+      Octokit::Client,
+      user: instance_double('User', name: 'test'),
+      'auto_paginate=': nil
+    )
+  end
+
+  def fixture(file)
+    path = File.join(File.dirname(__FILE__), 'test-data', 'github_branch_protection', file)
+    JSON.parse(File.read(path), symbolize_names: true)
+  end
+
+  def stub_response(hash)
+    stubs = Faraday::Adapter::Test::Stubs.new
+    agent = Sawyer::Agent.new 'http://release-toolkit.com/specs' do |conn|
+      conn.builder.handlers.delete(Faraday::Adapter::NetHttp)
+      conn.adapter :test, stubs
+    end
+    Sawyer::Resource.new(agent, hash)
```

Comment:
Borrowed directly [from the `sawyer` gem's own tests](https://github.com/lostisland/sawyer/blob/master/test/resource_test.rb#L7-L11)